File: 1.两数之和/main.py
# coding=utf-8

class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        hash = {}

        for i in range(len(nums)):
            if target - nums[i] not in hash:
                hash[nums[i]] = i
            else:
                return [hash[target - nums[i]], i]
        return []


if __name__ == '__main__':
    a = [2, 7, 11, 15]
    S=Solution()
    print(S.twoSum(nums=a, target=9))

File: 1.两数之和/main.js

/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function (nums, target) {
    let map = {}
    for (var i = 0; i < nums.length; i++) {
        dis = target - nums[i]
        if (map[dis] == undefined) {
            map[nums[i]] = i
        } else {
            return [map[dis], i]
        }
    }
    return []
};
(function main() {
    console.log(twoSum([2, 7, 9, 11], 14));
}());